src/zenml/integrations/github/orchestrators/github_actions_orchestrator.py

```diff
@@ -0,0 +1,460 @@
+
+
+class GitHubActionsOrchestrator(BaseOrchestrator):
+    """Orchestrator responsible for running pipelines using GitHub Actions.
+
+    Attributes:
+        custom_docker_base_image_name: Name of a docker image that should be
+            used as the base for the image that will be run on GitHub Action
+            runners. If no custom image is given, a basic image of the active
+            ZenML version will be used. **Note**: This image needs to have
+            ZenML installed, otherwise the pipeline execution will fail. For
+            that reason, you might want to extend the ZenML docker images
+            found here: https://hub.docker.com/r/zenmldocker/zenml/
+        prevent_dirty_repository: If `True`, this orchestrator will raise an
+            exception when trying to run a pipeline while there are still
+            untracked/uncommitted files in the git repository.
+        push: If `True`, this orchestrator will automatically commit and push
+            the GitHub workflow file when running a pipeline. If `False`, the
+            workflow file will be written to the correct location but needs to
+            be committed and pushed manually.
+    """
+
+    custom_docker_base_image_name: Optional[str] = None
+    prevent_dirty_repository: bool = True
+    push: bool = False
+
+    _git_repo: Optional[Repo] = None
+
+    # Class configuration
+    FLAVOR: ClassVar[str] = GITHUB_ORCHESTRATOR_FLAVOR
+
+    @property
+    def git_repo(self) -> Repo:
+        """Returns the git repository for the current working directory.
+
+        Raises:
+            RuntimeError: If there is no git repository for the current working
+                directory or the repository remote is not pointing to GitHub.
+        """
+        if not self._git_repo:
+            try:
+                self._git_repo = Repo(search_parent_directories=True)
+            except InvalidGitRepositoryError:
+                raise RuntimeError(
+                    "Unable to find git repository in current working "
+                    f"directory {os.getcwd()} or its parent directories."
+                )
+
+            remote_url = self.git_repo.remote().url
+            is_github_repo = any(
+                remote_url.startswith(prefix)
+                for prefix in GITHUB_REMOTE_URL_PREFIXES
+            )
+            if not is_github_repo:
+                raise RuntimeError(
+                    f"The remote URL '{remote_url}' of your git repo "
+                    f"({self._git_repo.git_dir}) is not pointing to a GitHub "
+                    "repository. The GitHub Actions orchestrator runs "
+                    "pipelines using GitHub Actions and therefore only works "
+                    "with GitHub repositories."
+                )
+
+        return self._git_repo
+
+    @property
+    def workflow_directory(self) -> str:
+        """Returns path to the github workflows directory."""
+        assert self.git_repo.working_dir
```

Comment:
Could we improve this by raising an error with a informative message rather than asserting?